src/ElectricityMarketData.jl

```diff
@@ -1,5 +1,64 @@
 module ElectricityMarketData
 
-# Write your package code here.
+using Dates
+using Logging
+using Tables
 
+export ElectricityMarket, list_markets, available_time_series, get_rt_lmp_raw_data, rt_lmp_data
+
+"""
+    ElectricityMarket
+
+Abstract type representing an electricity market.
+"""
+abstract type ElectricityMarket end
+
+"""
+    list_markets() :: Vector{ElectricityMarket}
+
+Return a vector of all available electricity markets.
+"""
+function list_markets() :: Vector{Symbol}
+    return Vector{Symbol}()
+end
+
+"""
+    available_time_series(market::ElectricityMarket) :: Vector{NamedTuple}
+
+Return Vector of `NamedTuple` of available time series for the given `market`.
+
+Ex:
+```
+[
+    (name="RT-load", unit="MW", resolution=Hour(1), first_date=DateTime("2021-01-01T00:00:00"), method=rt_load_data, description="Real-time load data"),
+    (name="RT-LMP", unit="MWh", resolution=Hour(1), first_date=DateTime("2021-01-01T00:00:00"), method=rt_lmp_data, description="Real-time Locational Marginal Price data"),
+]
+```
+"""
+function available_time_series(market::ElectricityMarket)
+    @warn "No time series registered for $(market)"
+    return Vector{NamedTuple}()
 end
+
+"""
+    get_rt_lmp_raw_data(market::ElectricityMarket, start_date::DateTime, end_date::DateTime; folder::AbstractString=mktempdir(), parser::Function=(args...) -> nothing)
+
+Download raw data for Real-Time (RT) Locational Marginal Price (LMP) for the given `market` and `start_date` to `end_date` and save it in `folder`.
+Parse the data using `parser` if provided.
+"""
+function get_rt_lmp_raw_data(market::ElectricityMarket, start_date::DateTime, end_date::DateTime; folder::AbstractString="", parser::Function=(args...) -> nothing)
+    throw(MethodError(get_rt_lmp_raw_data, (market, DateTime, DateTime)))
+end
+
+"""
+    rt_lmp_data(market::ElectricityMarket, start_date::DateTime, end_date::DateTime; folder::AbstractString=mktempdir(), parser::Function=(args...) -> nothing) :: Tables.AbstractTable
+
+Return a table with Real-Time (RT) Locational Marginal Price (LMP) data for the given `market` and `start_date` to `end_date`.
+Parse the data using `parser` if provided.
+If the data is not available, download it and save it in `folder`. 
+"""
+function rt_lmp_data(market::ElectricityMarket, start_date::DateTime, end_date::DateTime; folder::AbstractString="", parser::Function=(args...) -> nothing)
```

Comment:
Not a big fan of **​​folder::AbstractString=""** too
1) I believe that in most cases, users will not want to use the files directly
2) To use as a cache, we will not be able to delete the files after the call, that is, the user will have to do this
3) If the user does not pay attention to this parameter, we will create folders and files that may conflict with their file tree.

I suggest:
1) delete this parameter and always use the OS's temporary directory tempdir() (this way it is hidden and we can use it as a cache, the OS will delete it periodically)
2) Leave **nothing** as the default value, in this case, we use the temporary directory, if the user set a value, we use folder


https://docs.julialang.org/en/v1/base/file/#Base.Filesystem.tempdir

Reply:
I agree, I just set as "" here to test the through error without the need to create a folder.

I try to avoid the default temp dir and prefer creating new temp dirs (mktempdir) to avoid any possible conflicts. Nevertheless, I agree that it would defeat the purpose of a possible cache here. How about: https://pkgdocs.julialang.org/v1/artifacts/ ?

Reply:
I will change to tempdir for now